src/core/column/countna.h

```diff
@@ -0,0 +1,64 @@
+//------------------------------------------------------------------------------
+// Copyright 2023 H2O.ai
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//------------------------------------------------------------------------------
+#ifndef dt_COLUMN_COUNTNA_h
+#define dt_COLUMN_COUNTNA_h
+#include "column/reduce_unary.h"
+namespace dt {
+
+
+template <typename T, typename U, bool COUNT_NOT_NULL, bool COUNT_ALL_ROWS, bool IS_GROUPED>
```

Comment:
We only have two functions that are implemented here, i.e. `count()` or `countna()`, why do we need two booleans to handle that? If we introduce one boolean `COUNTNA`, then
- for `COUNTNA == 0` we implement `count()` logic;
- for `COUNTNA == 1` we implement `countna()` logic.

Reply:
One Boolean is to get count of rows for valid elements, the second Boolean gets count of all rows and does not consider element validity. I renamed the booleans for more clarity. Open to suggestions on renaming and better implementation

Reply:
>the second Boolean gets count of all rows and does not consider element validity

Does it mean you want to implement a new functionality here? Because currently `count()` only calculates number of non-missing elements, while `countna()` returns the number of missings.

Reply:
I'm actually wondering what is the best way to return a number of rows per group when using `by()`. It seems that one has to do `count() + countna()` that is not really good. Is it something you wanted to address?

Reply:
We probably need to update docs https://datatable.readthedocs.io/en/latest/api/dt/count.html and make parameter `cols` to be optional.

Reply:
The implementation takes care of the internal `count0()` - this section of the code in `fexpr_count_countna.cc` takes care of that: 

```py
    Workframe evaluate_n(EvalContext &ctx) const override {
      Workframe outputs(ctx);
      Workframe wf = arg_->evaluate_n(ctx);
      Groupby gby = ctx.get_groupby();
      // this covers a scenario where
      // we dont care about the presence or absence of NAs
      // we just want the total number of rows
      bool count_all_rows = arg_->get_expr_kind() == Kind::None;

      if (count_all_rows && !gby) {
        auto value = static_cast<int64_t>(ctx.nrows());
        Column coli = Const_ColumnImpl::make_int_column(1, value, SType::INT64);
        outputs.add_column(std::move(coli), "count", Grouping::GtoONE);
        return outputs;
      }

      if (count_all_rows && gby) {
        Column coli = Column(new Latent_ColumnImpl(
                new CountAllRows_ColumnImpl(gby)
              ));
        outputs.add_column(std::move(coli), "count", Grouping::GtoONE);
        return outputs;
      }
```

In the code above, there is no addition of count and countna, it just grabs the total number of rows. Within a groupby it just does an i1 - 10 ( i basically reused the internal count0() implementation)

I have updated the docs to reflect that count of all rows is possible, irrespective of the presence of nulls. An example already existed to show that as well.

Reply:
Yep, exactly.